django/db/models/deletion.py

```diff
@@ -64,7 +64,7 @@ class Collector(object):
     def __init__(self, using):
         self.using = using
         # Initially, {model: {instances}}, later values become lists.
-        self.data = {}
+        self.data = OrderedDict()
```

Comment:
Since this change affects deletion, below are the results for delete related benchmarks. I ran it a couple of  times. It gives me `Not significant` all the times even if the min/avg values are slightly different from run to run.

```
Running 'query_delete' benchmark ...
Min: 0.001014 -> 0.001006: 1.0078x faster
Avg: 0.001322 -> 0.001322: 1.0002x slower
Not significant
Stddev: 0.00085 -> 0.00092: 1.0913x larger (N = 50)

Running 'query_delete_related' benchmark ...
Min: 0.001427 -> 0.001379: 1.0348x faster
Avg: 0.001756 -> 0.001748: 1.0042x faster
Not significant
Stddev: 0.00092 -> 0.00105: 1.1418x larger (N = 50)

Running 'model_delete' benchmark ...
Min: 0.000473 -> 0.000449: 1.0536x faster
Avg: 0.000550 -> 0.000506: 1.0878x faster
Not significant
Stddev: 0.00024 -> 0.00023: 1.0721x smaller (N = 50)
```